crates/optimism/chainspec/src/lib.rs

```diff
@@ -178,6 +188,58 @@ pub struct OpChainSpec {
     pub inner: ChainSpec,
 }
 
+impl OpChainSpec {
+    /// Read from parent to determine the base fee for the next block
+    pub fn next_block_base_fee(&self, parent: &Header, timestamp: u64) -> U256 {
+        let is_holocene = self.inner.is_fork_active_at_timestamp(
+            reth_optimism_forks::OptimismHardfork::Holocene,
+            timestamp,
+        );
+        // If we are in the Holocene, we need to use the base fee params from the parent block's
+        // nonce Else, use the base fee params from chainspec
+        if is_holocene {
+            match decode_holocene_1559_params(parent.extra_data.clone()) {
+                Ok((denominator, elasticity)) => {
+                    if elasticity == 0 && denominator == 0 {
+                        return U256::from(
+                            parent
+                                .next_block_base_fee(self.base_fee_params_at_timestamp(timestamp))
+                                .unwrap_or_default(),
+                        );
+                    }
+                    let base_fee_params =
+                        BaseFeeParams::new(denominator as u128, elasticity as u128);
+                    return U256::from(
+                        parent.next_block_base_fee(base_fee_params).unwrap_or_default(),
+                    )
+                }
+                Err(e) => {
+                    panic!("Error occurred: {}", e)
+                }
```

Comment:
we should not have any panics in the code, this method should return a `Result` as well

Reply:
Curious how should the caller method handle errors for this method?
https://github.com/paradigmxyz/reth/pull/11887/files#diff-113cda2d142bb7cdd2474ed9a9d9cfdfe39fbcf58dc764a85ca488880316ed6fR159

Doesn't seem ideal to also modify that to have `Result` return type as well since it's a `trait` method shared between multiple structs. But to not modify the return type it'd require either `unwrap()` or `panic` I think?

Reply:
hmm, we really shouldn't be panicking in the node at all though, so we probably have to change the calling method to return a result.

We could change the trait like this:
```
trait ConfigureEvmEnv {
    type Header;
    type NextCfgError;

    // methods here ...
    fn next_cfg_and_block_env(/* args */) -> Result<...>
```